### PR TITLE
RDS logging

### PR DESCRIPTION
**Pull request by LindsayYoung**
We want to get CloudWatch logs going, so  we need to configure a IAM Role and pass the ARN to the instances so that they can enable the enhanced monitoring.

### CODE REVIEW

File: terraform/fec_rds.tf [hunk 1]
```diff
@@ -16,6 +16,42 @@ provider "aws" {
   region = "${var.region}"
 }
 
+resource "aws_iam_role" "monitoring_role" {
+  name = "AmazonRDSEnhancedMonitoringRoleTF"
+
+  assume_role_policy = <<EOF
```
Comment (jmcarp) [inline terraform/fec_rds.tf]: It looks like this should be the role policy, not the assume role policy--the first is what the role is allowed to do, and the second is who is allowed to assume the role. I think you want to move this policy to a separate `aws_iam_role_policy` resource and associate the policy resource with the role: https://www.terraform.io/docs/providers/aws/r/iam_role_policy.html.

**Comment by LindsayYoung** [comment]
@jmcarp if you have a moment can you take look at this?

File: terraform/fec_rds.tf [hunk 2]
```diff
@@ -16,6 +16,61 @@ provider "aws" {
   region = "${var.region}"
 }
 
+# logging role
+aws_iam_role_policy = <<EOF
```
Comment (jmcarp) [inline terraform/fec_rds.tf]: The syntax here should be:

```hcl
resource "aws_iam_role_policy" "test_policy" {
  name = "test_policy"
  role = "${aws_iam_role.test_role.id}"
  policy = <<EOF
...
EOF
}
```

The docs at https://www.terraform.io/docs/providers/aws/r/iam_role_policy.html include a good example. You can also use the `terraform validate` command to check whether the configuration is valid.

**Comment by LindsayYoung** [comment]
Thanks @jmcarp the `terraform validate` is so useful!

File: terraform/fec_rds.tf [hunk 3]
```diff
@@ -16,6 +16,68 @@ provider "aws" {
...
+   }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role" "test_role" {
+  name = "test_role"
+
+  assume_role_policy = <<EOF
```
Comment (jmcarp) [inline terraform/fec_rds.tf]: This doesn't look like a policy that controls what can assume the role. Unless the docs for RDS monitoring say something different, I'm guessing you'll want the standard assume role policy from the docs:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "ec2.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
```

File: terraform/fec_rds.tf [hunk 4]
```diff
@@ -16,6 +16,68 @@ provider "aws" {
...
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+      ],
+      "Resource": [
+        "arn:aws:logs:*:*:*"
```
Comment (jmcarp) [inline terraform/fec_rds.tf]: Probably needs to be `arn:aws-us-gov:logs:*:*:*`.

File: terraform/fec_rds.tf (hunk 3 above)
Comment (LindsayYoung) [reply terraform/fec_rds.tf]: I was trying to mirror the way I set up the logging role I set up in the console, but if the standard policy is better, I can go with that.

Comment (jmcarp) [reply terraform/fec_rds.tf]: Looks like these policies are reversed--the `assume_role_policy` on the role should control `AssumeRole`, and the `policy` on the policy should control logging.

Comment (LindsayYoung) [reply terraform/fec_rds.tf]: Thanks for catching that, it seems obvious now that you say it.

File: terraform/fec_rds.tf [hunk 5]
```diff
@@ -146,6 +204,7 @@ resource "aws_db_instance" "rds_production" {
   maintenance_window = "Sat:06:00-Sat:08:00"
   parameter_group_name = "${aws_db_parameter_group.fec_default.id}"
   apply_immediately = true
+  test_role_arn = "${aws_iam_role.test_role.arn}"
```
Comment (jmcarp) [inline terraform/fec_rds.tf]: Looks like this variable name should be `monitoring_role_arn`.

**Review by ccostino — APPROVED** [review]
Thank you, @LindsayYoung, @jmcarp, and @LinuxBozo!

**Comment by LindsayYoung** [comment]
Thank you @jmcarp, @LinuxBozo and @ccostino!

**Pull request merged**